pgmpy/models/DynamicBayesianNetwork.py

```diff
@@ -0,0 +1,480 @@
+        some static nodes and is then replicated over a certain time slice.
+
+        The nodes that will be added to the network will have the string form.
+
+        However the hidden nodes will compulsory have the following form.
+        (node_name, time_slice)
+
+        Here, node_name is a string while the time_slice is an integer value, which denotes
+        the index of the time_slice that the node belongs to.
+
+        Edges are represented as links between the nodes.
+
+        Parameters:
+        ----------
+        data: Data to initialize graph.  If data=None (default) an empty
+              graph is created.  The data can be an edge list, or any
+              NetworkX graph object
+
+        Examples:
+        --------
+        Create an empty Dynamic Bayesian Network with no nodes and no edges
+        >>> from pgmpy.models import DynamicBayesianNetwork as DBN
+        >>> dbn = DBN()
+        adding nodes and edges inside the dynamic bayesian network. A single
+        node can be added using the method below.
+        >>> dbn.add_nodes_from(['D','G','I','S','L'])
+        >>> dbn.add_edges_from([(('D',0),('G',0)),(('I',0),('G',0)),(('G',0),('L',0))])
+
+        most of the methods will be imported from Bayesian Model
+        However there will be a few changes
+        >>> dbn.list_of_nodes
+        ['D','G','I','S','L']
+        >>> dbn.edges()
+        [(('D',0),('G',0)),(('I',0),('G',0)),(('G',0),('L',0))]
+        If some edges connect nodes not yet in the model, the nodes
+        are added automatically.  There are no errors when adding
+        nodes or edges that already exist.
+        Methods:
+        -------
+        add_nodes_from
+        add_edge
+        add_edges_from
+        intra_slice
+        inter_slice
+        add_cpds
+        get_cpds
+        compute_initial_state
+        unroll
+        remove_timeslice
+        """
+        super().__init__()
+        self.inter_edges = []
+        self.intra_edges = []
+        self.dbn_cpds = OrderedDict()
+        self.list_of_nodes = []
+        if ebunch:
+            self.add_nodes_from(ebunch)
+        self.timestate = 1
+        self.computed = False
+
+    def add_nodes_from(self, nodes):
+        """
+        adding nodes in the static Bayesian Network.
+        Examples
+        --------
+        >>> from pgmpy.models import DynamicBayesianNetwork as DBN
+        >>> dbn = DBN()
+        >>> dbn.add_nodes_from(['D','G','I','S','L'])
+        """
+        for node in nodes:
+            if not isinstance(node, str):
+                raise ValueError('the node must be of the string type')
+            self.list_of_nodes.extend(node)
+
+    def add_edge(self, start, end, **kwargs):
+        """
+        Add an edge between two nodes.
+
+        The nodes will be automatically added if they are
+        not already in the initial static Bayesian Network.
+
+        Parameters
+        ----------
+        start: the starting node of the edge
```

Comment:
@palashahuja Have a look at other methods to see the proper format of parameters